widget/window_switcher.lua

```diff
@@ -0,0 +1,390 @@
+                                    forced_width = name_forced_width,
+                                    valign  = name_valign,
+                                    id = "text_role",
+                                    widget = wibox.widget.textbox
+                                },
+                                margins = name_margins,
+                                widget = wibox.container.margin
+                            },
+                            spacing = client_icon_horizontal_spacing,
+                            layout = wibox.layout.fixed.horizontal
+                        },
+                        layout = wibox.layout.flex.vertical
+                    }
+                }
+            }
+        end
+
+        return awful.widget.tasklist {
+            screen = s,
+            filter = awful.widget.tasklist.filter.currenttags,
+            buttons = mouse_keys,
+            style = { font = name_font, fg_normal = name_normal_color, fg_focus = name_focus_color },
+            layout = { layout  = wibox.layout.fixed.vertical, spacing = clients_spacing },
+            widget_template =
+            {
+                widget = wibox.container.background,
+                id = "bg_role",
+                forced_width = client_width,
+                forced_height = client_height,
+                create_callback = function(self, c, _, __)
+                    if (font_icons) ~= nil then
+                        set_font_icon(self, c)
+                        c:connect_signal("property::class", function() set_font_icon(self, c) end)
+                    elseif (custom_icons) ~= nil then
+                        set_custom_icon(self, c)
+                        c:connect_signal("property::class", function() set_custom_icon(self, c) end)
+                    end
+                end,
+                {
+                    icon_widget(),
+                    {
+                        {
+                            forced_width = name_forced_width,
+                            valign  = name_valign,
+                            id = "text_role",
+                            widget = wibox.widget.textbox
+                        },
+                        margins = name_margins,
+                        widget = wibox.container.margin
+                    },
+                    spacing = client_icon_horizontal_spacing,
+                    layout = wibox.layout.fixed.horizontal
+                },
+            },
+        }
+    end
+
+    s.window_switcher_box = awful.popup
+    ({
+        bg = "#00000000",
+        visible = false,
+        ontop = true,
+        placement = awful.placement.centered,
+        screen = s,
+        widget =
+        {
+            {
+                tasklist_widget(),
+                margins = client_margins,
+                widget = wibox.container.margin
+            },
+            border_width = border_width,
+            border_color = border_color,
```

Comment:
Do `shape_border_color` and `shape_border_width` here instead for stable compatibility